storage/tests/unit/test_blob.py

```diff
@@ -383,6 +383,35 @@ def test_generate_signed_url_w_credentials(self):
         credentials = object()
         self._basic_generate_signed_url_helper(credentials=credentials)
 
+    def test_generate_signed_url_non_ascii(self):
+        BLOB_NAME = u'Caf\xe9.txt'
+        EXPIRATION = '2014-10-16T20:34:37.000Z'
+        connection = _Connection()
+        client = _Client(connection)
+        bucket = _Bucket(client)
+        blob = self._make_one(BLOB_NAME.encode('utf-8'), bucket=bucket)
```

Comment:
There is no need to encode the name here:  the `Blob` constructor just decodes it if it is bytes.